File: template/app/src/main/java/com/xmartlabs/template/ui/recyclerfragmentexample/RecyclerExampleAdapter.java
package com.xmartlabs.template.ui.recyclerfragmentexample;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.xmartlabs.template.R;

import java.util.List;

class RecyclerExampleAdapter extends RecyclerView.Adapter<RecyclerExampleAdapter.RecyclerExampleViewHolder> {
  @NonNull
  private List<String> items;

  RecyclerExampleAdapter(@NonNull List<String> strings) {
    items = strings;
  }

  @Override
  public RecyclerExampleViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
    View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_recycler_example, parent, false);
    return new RecyclerExampleViewHolder(view);
  }

  @Override
  public void onBindViewHolder(RecyclerExampleViewHolder holder, int position) {
    holder.bind(items.get(position));
  }

  @Override
  public int getItemCount() {
    return items.size();
  }

  static class RecyclerExampleViewHolder extends RecyclerView.ViewHolder {
    RecyclerExampleViewHolder(@NonNull View view) {
      super(view);
    }

    void bind(@NonNull String string) {
      ((TextView) itemView).setText(string);
    }
  }
}

File: template/app/src/main/java/com/xmartlabs/template/ui/recyclerfragmentexample/RecyclerExampleAdapter.kt
package com.xmartlabs.template.ui.recyclerfragmentexample

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView

import com.xmartlabs.template.R

internal class RecyclerExampleAdapter(private val items: List<String>)
  : RecyclerView.Adapter<RecyclerExampleAdapter.RecyclerExampleViewHolder>() {

  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerExampleViewHolder {
    val view = LayoutInflater.from(parent.context).inflate(R.layout.item_recycler_example, parent, false)
    return RecyclerExampleViewHolder(view)
  }

  override fun onBindViewHolder(holder: RecyclerExampleViewHolder, position: Int) = holder.bind(items[position])

  override fun getItemCount() = items.size

  internal class RecyclerExampleViewHolder(view: View) : RecyclerView.ViewHolder(view) {
    fun bind(string: String) = (itemView as? TextView)?.let { it.text = string } ?: Unit
  }
}
